test/OpenApiTests/Headers/HeaderTests.cs

```diff
@@ -0,0 +1,146 @@
+using System.Text.Json;
+using FluentAssertions;
+using TestBuildingBlocks;
+using Xunit;
+
+namespace OpenApiTests.Headers;
+
+public sealed class HeaderTests : IClassFixture<OpenApiTestContext<OpenApiStartup<HeadersDbContext>, HeadersDbContext>>
+{
+    private readonly OpenApiTestContext<OpenApiStartup<HeadersDbContext>, HeadersDbContext> _testContext;
+
+    public HeaderTests(OpenApiTestContext<OpenApiStartup<HeadersDbContext>, HeadersDbContext> testContext)
+    {
+        _testContext = testContext;
+
+        testContext.UseController<CountriesController>();
+
+        testContext.SwaggerDocumentOutputDirectory = "test/OpenApiEndToEndTests/Headers";
+    }
+
+    [Theory]
+    [InlineData("/countries.get")]
+    [InlineData("/countries.head")]
+    [InlineData("/countries/{id}.get")]
+    [InlineData("/countries/{id}.head")]
+    [InlineData("/countries/{id}/languages.get")]
+    [InlineData("/countries/{id}/languages.head")]
+    [InlineData("/countries/{id}/relationships/languages.get")]
+    [InlineData("/countries/{id}/relationships/languages.head")]
+    public async Task Get_and_head_endpoints_have_caching_headers(string endpointPath)
```

Comment:
I don't think the test name should include the inputs it is being used with.
```suggestion
    public async Task Endpoints_have_caching_headers(string endpointPath)
```